keras/src/callbacks/memory_usage_callback.py

```diff
@@ -0,0 +1,192 @@
+import os
+import warnings
+
+from keras.src import backend as K
+from keras.src.api_export import keras_export
+from keras.src.callbacks.callback import Callback
+
+# Attempt to import psutil for CPU memory monitoring
+try:
+    import psutil
+except ImportError:
+    psutil = None
+
+
+@keras_export("keras.callbacks.MemoryUsageCallback")
+class MemoryUsageCallback(Callback):
+    """Monitors and logs memory usage
+    (CPU + optional GPU/TPU/OpenVINO) during training.
+
+    This callback measures:
+
+      - **CPU**: via psutil.Process().memory_info().rss
+      - **GPU/TPU**: via backend‐specific APIs
+        (TensorFlow, PyTorch, JAX)
+
+    Logs are printed to stdout at the start/end of each epoch and,
+    if `log_every_batch=True`, after every batch.
+    If `tensorboard_log_dir` is provided, scalars are also written
+    via `tf.summary` (TensorBoard).
+
+    Args:
+        monitor_gpu (bool): If True, attempt to measure accelerator memory.
+        log_every_batch (bool): If True, also log after each batch.
```

Comment:
what is the default behavior? 
Log at the end of each epoch? - You would document  default behavior in docstring